compression/extractor.go

```diff
@@ -0,0 +1,76 @@
+/*
```

Comment:
I feel we should more have a single `zinfo` interface which has 1 impl for each compression (e.g. GzipZinfo).